generator/k8s_client_template.go

```diff
@@ -26,13 +29,34 @@ func NewForConfig(cfg rest.Config) (Interface, error) {
 	if err := {{.prefix}}AddToScheme(scheme); err != nil {
 		return nil, err
 	}
-	controllerFactory, err := controller.NewSharedControllerFactoryFromConfig(&cfg, scheme)
+	var sharedOpts *controller.SharedControllerFactoryOptions
+	if skipUpdateOnNoChange() {
+		sharedOpts = &controller.SharedControllerFactoryOptions{
+			SkipNoChangeUpdate: true,
+		}
+	}
+	controllerFactory, err := controller.NewSharedControllerFactoryFromConfigWithOptions(&cfg, scheme, sharedOpts)
 	if err != nil {
 		return nil, err
 	}
 	return NewFromControllerFactory(controllerFactory)
 }
 
+func skipUpdateOnNoChange() bool {
```

Comment:
If I'm not mistaken - this function is going to be in all generated clients, despite the fact that it is identical in every case (i.e. no properties of the template seem to be used in the function). 

Would it be better to place this in some part of the library which isn't generated and configure the template to use that library function rather than re-generating the function for every client?

Reply:
Fixed to do this. I avoided it before because the files produced by the template will be in a different application so I thought it would be awkward but it looks like the template already references norman packages so shouldn't be an issue.